File: tests/testdata/codegen/empty-list-property-pp/go/empty-list-property.go
package main

import (
	"github.com/khulnasoft/codeinfra-azure-native/sdk/go/azure/storage"
	"github.com/khulnasoft/codeinfra/sdk/v3/go/codeinfra"
)

func main() {
	codeinfra.Run(func(ctx *codeinfra.Context) error {
		_, err := storage.NewStorageAccount(ctx, "storageAccounts", &storage.StorageAccountArgs{
			AccountName:       codeinfra.String("sto4445"),
			Kind:              codeinfra.String(storage.KindBlockBlobStorage),
			Location:          codeinfra.String("eastus"),
			ResourceGroupName: codeinfra.String("res9101"),
			Sku: &storage.SkuArgs{
				Name: codeinfra.String(storage.SkuName_Premium_LRS),
			},
			NetworkRuleSet: &storage.NetworkRuleSetArgs{
				DefaultAction: storage.DefaultActionAllow,
				IpRules:       storage.IPRuleArray{},
			},
		})
		if err != nil {
			return err
		}
		return nil
	})
}

File: tests/testdata/codegen/empty-list-property-pp/dotnet/empty-list-property.cs
using System.Collections.Generic;
using System.Linq;
using Codeinfra;
using AzureNative = Codeinfra.AzureNative;

return await Deployment.RunAsync(() => 
{
    var storageAccounts = new AzureNative.Storage.StorageAccount("storageAccounts", new()
    {
        AccountName = "sto4445",
        Kind = AzureNative.Storage.Kind.BlockBlobStorage,
        Location = "eastus",
        ResourceGroupName = "res9101",
        Sku = new AzureNative.Storage.Inputs.SkuArgs
        {
            Name = AzureNative.Storage.SkuName.Premium_LRS,
        },
        NetworkRuleSet = new AzureNative.Storage.Inputs.NetworkRuleSetArgs
        {
            DefaultAction = AzureNative.Storage.DefaultAction.Allow,
            IpRules = new() { },
        },
    });

});

